zipkin/src/main/java/zipkin2/internal/DelayLimiter.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright 2015-2019 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package zipkin2.internal;
+
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.DelayQueue;
+import java.util.concurrent.Delayed;
+import java.util.concurrent.TimeUnit;
+
+/** Limits invocations of a given context to at most once per period. */
+// this is a dependency-free variant formerly served by an expiring guava cache
+public final class DelayLimiter<C> {
+
+  public static <C> DelayLimiter<C> create() {
+    return new Builder().build();
+  }
+
+  public static Builder newBuilder() {
+    return new Builder();
+  }
+
+  public static final class Builder {
```

Comment:
Can we add a Ticker here and inject it so no need for the powermocks?

Reply:
ticker is a guava type.. if it is really important to not use powermock we can..
* make DelayLimiter not final and expose a nanoTime method overridden in tests
* add a custom Ticker type only used in tests

In either case it would only be used in tests, as this is an internal type anyway. Probably the least messy and class bloat way would be to make an overridable method. otherwise we can leave it.

Reply:
going with overridable method for now as I also don't like the powermock cruft

Reply:
now the tests run much much faster :)